### PR TITLE
Python 2 Workflow Compatibility 

### PR DESCRIPTION
**Pull request by moppr**
This merge fixes the following issue: https://github.com/cadop/pyCGM/issues/42

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -3,8 +3,15 @@
...
-    long_description = fh.read()
+vers = sys.version_info.major
+if vers == 2:
+    with open("README.md", "r") as fh:
+        long_description = fh.read()
+else:
+    with open("README.md", "r",encoding="utf8") as fh:
+        long_description = fh.read()
+
+requires = ">=2, <3" if vers == 2 else ">=3"
```
Comment (cadop) [inline setup.py:14]: I am not sure I understand the point of this.

File: pyCGM_Single/pycgmIO.py [hunk 2]
```diff
@@ -879,29 +893,29 @@ def writeResult(data,filename,**kargs):
...
-        >>> result = lines[7].strip().split(',') 
+        >>> result = lines[7].strip().split(',')
         >>> result #doctest: +NORMALIZE_WHITESPACE
-        ['0.000000000000000', 
+        ['0.000000000000000',
          '-0.308494914509454', '-6.121292793370010', '7.571431102151710',...]
 
         Writing axis only.
 
-        >>> writeResult(data, outfile, angles=False, axis=True)
```
Comment (cadop) [inline pyCGM_Single/pycgmIO.py:897]: I don't think these changes are relevant.

File: pyCGM_Single/Pipelines.py [hunk 3]
```diff
@@ -53,9 +53,9 @@ def butterFilter(data, cutoff, Fs):
     First, we create a sin wave and add noise to it.
 
     >>> from numpy import arange, around, pi, random, shape, sin
-    >>> Fs = 360
+    >>> Fs = 360.0
     >>> t = 1
-    >>> x = arange(0,t,1/Fs)
+    >>> x = arange(0,2,1/Fs)
```
Comment (cadop) [inline pyCGM_Single/Pipelines.py:58]: Why did you change this?

**Review by cadop — CHANGES_REQUESTED** [review]

File: pyCGM_Single/Pipelines.py (hunk 3 above)
Comment (moppr) [reply pyCGM_Single/Pipelines.py:58]: The 360.0 was intentional to fix a division by 0 error that was only happening on the Python 2 build, the t -> 2 is an error and will be removed.

File: setup.py (hunk 1 above)
Comment (moppr) [reply setup.py:14]: It's related to the line `python_requires='>=3',` inside the setuptools.setup() function. When that line is present, the build will fail on the "Install package" step, with the error `ERROR: Package 'pycgm' requires a different Python: 2.7.18 not in '>=3'`. This is because the package is being told that it requires a python version of at least 3.0, when it's running on 2.7. My change avoids this error (and doesn't appear to cause any issues because the rest of the workflow passes), but if there's a better way to handle the error then we can do so instead.

Comment (cadop) [reply setup.py]: what is wrong with `python_requires='2.7, >=3'` ?

Comment (moppr) [reply setup.py]: Commas in python_requires work as a logical and, so `>=2.7, >=3.6` would still be an issue for 2.7.

The [suggested solution](https://packaging.python.org/guides/dropping-older-python-versions/) to this problem (requiring 2.7 or 3.x+) is to first require 2.7+ and then exclude whatever 3.x versions are unsupported, such as `'>2.7, !=3.0.*, !=3.1.*, !=3.2.*, !=3.3.*'` if doing 2.7 and 3.4+.

I'm assuming based on your reply that the first sub-version of 3 that is compatible is 3.6, so I'll go ahead and implement the above method and mark this again.

Comment (moppr) [reply setup.py]: One more reference for good measure https://twitter.com/hynek/status/963711338500558850

Comment (cadop) [reply setup.py]: I am not asking how commas work. I am asking what the point line 14 is.

Comment (moppr) [reply setup.py]: That line isn't necessary with the newer solution I found and in fact was already removed in my most recent commit.

File: setup.py [hunk 4]
```diff
@@ -21,7 +22,7 @@
         "License :: OSI Approved :: MIT",
         "Operating System :: OS Independent",
     ],
-    python_requires='>=3',
+    python_requires='>=2.7, !=3.0, !=3.1, !=3.2, !=3.3, !=3.4, !=3.5',
```
Comment (cadop) [inline setup.py:25]: This can be simplified to 2.7 since there is no 2.8, and >=3.6

**Review by cadop — CHANGES_REQUESTED** [review]

**Review by cadop — APPROVED** [review]

**Comment by cadop** [comment]
closes https://github.com/cadop/pyCGM/issues/42
